### PR TITLE
Release: Adds changes for 3.20.0 release

### PR DESCRIPTION
**Pull request by ealsur**
## Contract changes in 3.20.0

Highlight: Change Feed pull model GA

image

## Contract changes in 3.20.0-preview

Highlight: ChangeFeedProcessor manual checkpoint contract update and Change Feed pull model GA (out of preview)

image

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -3,6 +3,43 @@ Preview features are treated as a separate branch and will not be included in th
...
+- [#2510](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2510) Query: Fixes handling of pipeline execution on partition merge. Introduced in [#2084](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2084).
+- [#2547](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2547) Query: Fixes incorrect order by query when the field is an object or array
+- [#2511](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2511) Availability: Fixes get account info retry logic to not go to secondary regions on 401(Forbidden)
+- [#2512](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2512) Caches: Fixes the cache to remove values if generator throws an exception. Thanks @johngallardo.
+- [#2516](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2516) Diagnostics: Fixes a race condition causing InvalidOperationException.
+- [#2530](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2530) Gateway: Fixes container recreate scenarios for Gateway Mode in session consistency
```
Comment (j82w) [inline changelog.md:20]: Please add the PR the regression was introduced and the version of the SDK.

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM except for the comment Jake made about adding the PR which introduced the regressions

File: changelog.md [hunk 2]
```diff
@@ -3,6 +3,43 @@ Preview features are treated as a separate branch and will not be included in th
...
 
+### <a name="3.20.0"/> [3.20.0](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.20.0) - 2021-06-18
+
+#### Added
+- [#2509](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2509) Change Feed: Adds change feed iterator APIs on containers
+- [#2558](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2558) Diagnostics: Adds Duration field to HttpResponseStatistics in Diagnostics
+- [#2491](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2491) Change Feed Processor: Adds support for Graph API accounts. Graph API accounts can now create lease containers with `/partitionKey` instead of `/id`.
+
+#### Fixed
+- [#2510](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2510) Query: Fixes InvalidOperationException when partitions are merged. Introduced in [#2084](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2084).
```
Comment (j82w) [inline changelog.md:14]: Follow the previous examples where it includes the SDK version it was introduced "Introduced in 3.15.0 PR [#1933](...)"

File: changelog.md [hunk 3]
```diff
@@ -3,6 +3,43 @@ Preview features are treated as a separate branch and will not be included in th
...
+- [#2510](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2510) Query: Fixes handling of pipeline execution on partition merge. Introduced in [#2084](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2084).
+- [#2547](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2547) Query: Fixes incorrect order by query when the field is an object or array
+- [#2511](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2511) Availability: Fixes get account info retry logic to not go to secondary regions on 401(Forbidden)
+- [#2512](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2512) Caches: Fixes the cache to remove values if generator throws an exception. Thanks @johngallardo.
+- [#2516](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2516) Diagnostics: Fixes a race condition causing InvalidOperationException.
```
Comment (j82w) [inline changelog.md:19]:
```suggestion
- [#2516](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2516) Diagnostics: Fixes a race condition causing InvalidOperationException. Introduced in 3.17.0 PR [#2097](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2097)
```

File: changelog.md [hunk 4]
```diff
@@ -3,6 +3,43 @@ Preview features are treated as a separate branch and will not be included in th
...
+- [#2511](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2511) Availability: Fixes get account info retry logic to not go to secondary regions on 401(Forbidden)
+- [#2512](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2512) Caches: Fixes the cache to remove values if generator throws an exception. Thanks @johngallardo.
+- [#2516](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2516) Diagnostics: Fixes a race condition causing InvalidOperationException. Introduced in 3.17.0 PR [#2097](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2097)
+- [#2530](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2530) Gateway: Fixes container recreate scenarios for Gateway Mode in session consistency. Introduced in 3.18.0 PR [#2165](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2165)
+- [#2552](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2552) UserAgent: Fixes UserAgent to have the correct number of clients
+- [#2562](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2562) Diagnostics: Fixes NullReferenceException caused by RegionsContacted in IClientSideRequestStatistics
```
Comment (j82w) [inline changelog.md:22]:
```suggestion
- [#2562](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2562) Diagnostics: Fixes NullReferenceException in service unavailable scenarios. Introduced in 3.18.0 PR [#2312](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2312)
```

File: changelog.md [hunk 5]
```diff
@@ -3,6 +3,43 @@ Preview features are treated as a separate branch and will not be included in th
...
```
Comment (j82w) [inline changelog.md:41]:
```suggestion
- [#2562](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2562) Diagnostics: Fixes NullReferenceException in service unavailable scenarios. Introduced in 3.18.0 PR [#2312](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2312)
```

File: changelog.md [hunk 6]
```diff
@@ -3,6 +3,43 @@ Preview features are treated as a separate branch and will not be included in th
...
+
+### <a name="3.20.0-preview"/> [3.20.0-preview](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.20.0-preview) - 2021-06-18
+
+#### Added
+- [#2509](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2509) Change Feed: Adds change feed iterator APIs on containers
+- [#2558](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2558) Diagnostics: Adds Duration field to HttpResponseStatistics in Diagnostics
+- [#2491](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2491) Change Feed Processor: Adds support for Graph API accounts. Graph API accounts can now create lease containers with `/partitionKey` instead of `/id`.
+- [#2488](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2488) Change Feed Processor: Refactors checkpoint API introduced in [#2331](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2331). This is considered a breaking change on the preview API as it evolves into GA.
```
Comment (j82w) [inline changelog.md:30]:
```suggestion
- [#2488](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2488) Change Feed Processor: Refactors checkpoint API to throw exception. Introduced in [#2331](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2331). This is considered a breaking change on the preview API as it evolves into GA.
```

**Review by j82w — DISMISSED** [review]

**Review by FabianMeiswinkel — DISMISSED** [review]
LGTM

**Review by kirankumarkolli — APPROVED** [review]

**Review by j82w — APPROVED** [review]

**Pull request merged**